packages/bloc/lib/src/bloc_base.dart

```diff
@@ -39,12 +44,57 @@ abstract class ErrorSink implements Closable {
   void addError(Object error, [StackTrace? stackTrace]);
 }
 
+/// An interface for hooks of the [BlocBase].
+///
+/// Contains the `onChange` and `onError` hooks.
+abstract class BlocBaseHooks<State> {
```

Comment:
thoughts on `LifecycleObserver` and `BlocLifecycleObserver`? I'm not a fan of `hooks` because the term hooks is typically associated with `flutter_hooks`.